src/status_im/hardwallet/core.cljs

```diff
@@ -1843,19 +1843,21 @@
   (let [tag-was-lost? (= "Tag was lost." (:error error))]
     (fx/merge cofx
               (if tag-was-lost?
-                (fx/merge cofx
-                          {:db             (-> db
-                                               (assoc-in [:hardwallet :on-card-connected] :hardwallet/prepare-to-sign)
-                                               (assoc-in [:hardwallet :pin :status] nil)
-                                               (assoc-in [:signing/sign :keycard-step] :connect))
-                           :utils/show-popup {:title   (i18n/label :t/error)
-                                              :content (i18n/label :t/cannot-read-card)}}))
+                (fn [{:keys [db] :as cofx}]
+                  (fx/merge cofx
+                            {:db             (-> db
+                                                 (assoc-in [:hardwallet :on-card-connected] :hardwallet/prepare-to-sign)
+                                                 (assoc-in [:hardwallet :pin :status] nil)
+                                                 (assoc-in [:signing/sign :keycard-step] :connect))
+                             :utils/show-popup {:title   (i18n/label :t/error)
+                                                :content (i18n/label :t/cannot-read-card)}})))
               (if (re-matches pin-mismatch-error (:error error))
-                (fx/merge cofx
-                          {:db (-> db
-                                   (update-in [:hardwallet :pin] merge {:status      :error
-                                                                        :sign        []
-                                                                        :error-label :t/pin-mismatch})
-                                   (assoc-in [:signing/sign :keycard-step] :pin))}
-                          (get-application-info (get-pairing db) nil))
+                (fn [{:keys [db] :as cofx}]
+                  (fx/merge cofx
+                            {:db (-> db
+                                     (update-in [:hardwallet :pin] merge {:status      :error
+                                                                          :sign        []
+                                                                          :error-label :t/pin-mismatch})
+                                     (assoc-in [:signing/sign :keycard-step] :pin))}
+                            (get-application-info (get-pairing db) nil)))
                 (show-wrong-keycard-alert cofx true)))))
```

Comment:
isn't this one still wrong? it looks like it is part of the fx/merge a is passed an extra cofx?

Reply:
Yeah it's broken and (+ in one more place in the same file which wasn't shown in PR). Although original issue is fixed even with this bug. I didn't notice that, will fix in this pr

Reply:
fixed